RecoHGCal/TICL/plugins/PFTICLProducer.cc

```diff
@@ -95,7 +114,33 @@ void PFTICLProducer::produce(edm::StreamID, edm::Event& evt, const edm::EventSet
       reco::TrackRef ref(ticl_cand.trackPtr().id(), int(ticl_cand.trackPtr().key()), &evt.productGetter());
       candidate.setTrackRef(ref);
     }
-    candidate.setTime(ticl_cand.time(), ticl_cand.timeError());
+
+    // HGCAL timing as default values
+    auto time = ticl_cand.time();
+    auto timeE = ticl_cand.timeError();
+
+    if (candidate.charge()) {  // Check MTD timing availability
+      const bool assocQuality = (*trackTimeQualH)[candidate.trackRef()] > timingQualityThreshold_;
+      if (assocQuality) {
+        const auto timeHGC = time;
+        const auto timeEHGC = timeE;
+        const auto timeMTD = (*trackTimeH)[candidate.trackRef()];
+        const auto timeEMTD = (*trackTimeErrH)[candidate.trackRef()];
+
+        if (useTimingAverage_ && (timeEMTD > 0 && timeEHGC > 0)) {
+          // Compute weighted average between HGCAL and MTD timing
+          const auto invTimeESqHGC = pow(timeEHGC, -2);
+          const auto invTimeESqMTD = pow(timeEMTD, -2);
+          timeE = (invTimeESqHGC * invTimeESqMTD) / (invTimeESqHGC + invTimeESqMTD);
+          time = (timeHGC * invTimeESqHGC + timeMTD * invTimeESqMTD) * timeE;
+          timeE = sqrt(timeE);
+        } else if (timeEMTD > 0) {  // Ignore HGCal timing until it will be TOF corrected
+          time = timeMTD;
+          timeE = timeEMTD;
+        }
+      }
+    }
+    candidate.setTime(time, timeE);
```

Comment:
Hi, sorry for the late comment: suppose that the candidate is charged but the MTD assocQuality fails, then if the HGCAL time is != -99 the candidate will have a non TOF corrected value, correct?
If this is the case, this has to be corrected and the initialization value has to be passed.